lld/wasm/SyntheticSections.cpp

```diff
@@ -604,9 +611,11 @@ void ElemSection::writeBody() {
   writeUleb128(os, indirectFunctions.size(), "elem count");
   uint32_t tableIndex = ctx.arg.tableBase;
   for (const FunctionSymbol *sym : indirectFunctions) {
-    assert(sym->getTableIndex() == tableIndex);
+    if (sym != nullptr) {
+      assert(sym->getTableIndex() == tableIndex);
+    }
```

Comment:
Can you instead write this one line as  `assert(!sym || sym->getTableIndex() == tableIndex))`